### PR TITLE
Several new Python libraries

### PR DESCRIPTION
**Pull request by dgilman**
#### Description
This adds portfiles for several new python libraries, all required to run poetry.  The py-lockfile port was updated to its latest (final) version and added python3 support.

###### Type(s)
- [x] enhancement

###### Tested on
macOS 10.13.6 17G10021
Xcode 10.1 10B61

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL?

- [x] checked your Portfile with `port lint`?
- [x] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

**Comment by dgilman** [comment]
I can break these up into individual PRs if it'll help with CI / help get the changes landed.

File: python/py-CacheControl/Portfile [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+name                py-CacheControl
+version             0.12.6
+revision            0
+categories-append   devel
+platforms           darwin
+license             Apache-2
+supported_archs     noarch
+
+python.versions     27 34 35 36 37 38
```
Comment (pmetzger) [inline python/py-CacheControl/Portfile]: Should we have _new_ ports depending on Python 2.7?

File: python/py-cachy/Portfile [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+name                py-cachy
+version             0.3.0
+revision            0
+categories-append   devel
+platforms           darwin
+license             MIT
+supported_archs     noarch
+
+python.versions     27 34 35 36 37 38
```
Comment (pmetzger) [inline python/py-cachy/Portfile]: Should _new_ python ports specify 2.7? I think that's a bad idea.

File: python/py-CacheControl/Portfile (hunk 1 above)
Comment (mf2k) [reply python/py-CacheControl/Portfile]: I don't think we should. Only 37 and 38 should be here.

File: python/py-cachy/Portfile (hunk 2 above)
Comment (mf2k) [reply python/py-cachy/Portfile]: Agreed. Please only add 37 and 38.

**Comment by dgilman** [comment]
I will take this feedback into review and start opening individual pull requests.

**Comment by reneeotten** [comment]
as a follow-up to the mailinglist discussion: it's probably better to first establish whether the end-goal adding ```poetry``` is feasible. If not, there is little point in adding Python ports for its dependencies.

**Pull request closed without merging**